apprise/plugins/NotifyNtfy.py

```diff
@@ -0,0 +1,573 @@
+
+
+class NtfyMode(object):
+    """
+    Define Ntfy Notification Modes
+    """
+    # App posts upstream to the developer API on Ntfy's website
+    CLOUD = "cloud"
+
+    # Running a dedicated private Ntfy Server
+    PRIVATE = "private"
+
+
+NTFY_MODES = (
+    NtfyMode.CLOUD,
+    NtfyMode.PRIVATE,
+)
+
+
+class NtfyPriority(object):
+    """
+    Ntfy Priority Definitions
+    """
+    MAX = 'max'
+    HIGH = 'high'
+    NORMAL = 'default'
+    LOW = 'low'
+    MIN = 'min'
+
+
+NTFY_PRIORITIES = (
+    NtfyPriority.MAX,
+    NtfyPriority.HIGH,
+    NtfyPriority.NORMAL,
+    NtfyPriority.LOW,
+    NtfyPriority.MIN,
+)
+
+
+class NotifyNtfy(NotifyBase):
+    """
+    A wrapper for Ntfy Notifications
+    """
+
+    # The default descriptive name associated with the Notification
+    service_name = 'ntfy'
+
+    # The services URL
+    service_url = 'https://ntfy.sh/'
+
+    # Insecure protocol (for those self hosted requests)
+    protocol = 'ntfy'
+
+    # The default protocol
+    secure_protocol = 'ntfys'
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_ntfy'
+
+    # Default upstream/cloud host if none is defined
+    cloud_notify_url = 'https://ntfy.sh'
+
+    # Message time to live (if remote client isn't around to receive it)
+    time_to_live = 2419200
+
+    # if our hostname matches the following we automatically enforce
+    # cloud mode
+    __auto_cloud_host = re.compile(r'ntfy\.sh', re.IGNORECASE)
+
+    # Define object templates
+    templates = (
+        '{schema}://{topic}',
+        '{schema}://{host}/{targets}',
+        '{schema}://{host}:{port}/{targets}',
+        '{schema}://{user}@{host}/{targets}',
+        '{schema}://{user}@{host}:{port}/{targets}',
+        '{schema}://{user}:{password}@{host}/{targets}',
+        '{schema}://{user}:{password}@{host}:{port}/{targets}',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'host': {
+            'name': _('Hostname'),
+            'type': 'string',
+        },
+        'port': {
+            'name': _('Port'),
+            'type': 'int',
+            'min': 1,
+            'max': 65535,
+        },
+        'user': {
+            'name': _('Username'),
+            'type': 'string',
+        },
+        'password': {
+            'name': _('Password'),
+            'type': 'string',
+            'private': True,
+        },
+        'topic': {
+            'name': _('Topic'),
+            'type': 'string',
+            'map_to': 'targets',
+            'regex': (r'^[a-z0-9_-]{1,64}$', 'i')
+        },
+        'targets': {
+            'name': _('Targets'),
+            'type': 'list:string',
+        },
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'attach': {
+            'name': _('Attach'),
+            'type': 'string',
+        },
```

Comment:
We're missing `Filename`, which is the displayname/filename; it's fine if it's missing for now, but it should be easy to add. 

See https://ntfy.sh/docs/publish/#attach-file-from-a-url and https://ntfy.sh/docs/publish/#list-of-all-parameters

Reply:
`X-Filename` has been added in the last check for Attachments.  but for local file attachments as identified [here](https://ntfy.sh/docs/publish/#attach-local-file).  This still needs to be done.

I would advise maybe reviewing what is there and see if the rest is okay, and then we can start adding the `--attach` support to Apprise to push uploaded content next.

Reply:
Sounds great.